### PR TITLE
Tagging csv

### PR DESCRIPTION
**Pull request by samiyaakhtar**
- Got rid of param for number of images in training.py
- Created simple changes in images api to allow us to pull tag state info which saves us from repeated calls to get the same image (but different information)
- Generating tagging.csv alongside tagged.csv

### CODE REVIEW

**Review by andrebriggs — COMMENTED** [review]
Overall these are the changes that we want for training.py however TAGGED_OUTPUT, TAGGING_OUTPUT, and TAGGING_IMAGE_DIR need to come from the legacy config. The reason for this is to avoid having the same configuration twice. I'll leave some suggestions on how we can do that

**Comment by samiyaakhtar** [comment]
@andrebriggs  should we be passing path to the legacy config in training.py? That way we can call validate_config from there

File: train_vnext/training.py [hunk 1]
```diff
@@ -13,10 +13,10 @@
 
 CONFIG_PATH = os.environ.get('ALCONFIG', None)
```
Comment (andrebriggs) [inline train_vnext/training.py:15]: @samiyaakhtar we need to the [legacy config ](https://github.com/CatalystCode/active-learning-detect/blob/master/config.ini). Maybe we can add something like
```
LEGACY_CONFIG_PATH = os.environ.get('ALCONFIG_LEGACY', None)
```

File: train_vnext/training.py [hunk 2]
```diff
@@ -13,10 +13,10 @@
...
 
-def train(config, num_images):
+def train(config):
```
Comment (andrebriggs) [inline train_vnext/training.py:18]: @samiyaakhtar We can reference the legacy config somewhere. Building from the previous comment:
```
legacy_config = get_legacy_config(LEGACY_CONFIG_PATH)
...
legacy_config["tagged_output"]
legacy_config["tagging_output"]
legacy_config["image_dir"]
legacy_config["train_dir"]
...
```

We would need to import [this](https://github.com/CatalystCode/active-learning-detect/blob/master/train_vnext/validate_config.py) file

**Review by andrebriggs — CHANGES_REQUESTED** [review]

File: train_vnext/training.py (hunk 2 above)
Comment (samiyaakhtar) [reply train_vnext/training.py:18]: Is the legacy config the one in the root folder called config.ini?

**Comment by andrebriggs** [comment]
> @andrebriggs should we be passing path to the legacy config in training.py? That way we can call validate_config from there

Yeah that will need to be a requirement now. The example I gave was via environment var

File: train_vnext/training.py [hunk 3]
```diff
@@ -49,25 +50,25 @@ def download_images(imageURLs, file_location):
     print("Downloaded images into " + file_location)
 
 
-def download_data_for_training(config):
+def download_data_for_training(config, user_name, function_url):
```
Comment (andrebriggs) [inline train_vnext/training.py]: @samiyaakhtar are we still using "config"?

File: train_vnext/validate_config.py [hunk 4]
```diff
@@ -1,13 +1,27 @@
...
 
 class IllegalArgumentError(ValueError):
     pass
 
+def initialize_training_location(config):
+        file_tree = pathlib.Path(os.path.expanduser(
+            config.get("train_dir"))
+        )
+
+        if file_tree.exists():
```
Comment (andrebriggs) [inline train_vnext/validate_config.py:18]: @samiyaakhtar The training shell script does this [too](https://github.com/andrebriggs/active-learning-detect/blob/fcfbcabcaf75cae155eaa71c801187a76b02d49e/train_vnext/active_learning_train.sh#L48) but I think it is fine for us to do this as well

File: train_vnext/training.py [hunk 5]
```diff
@@ -166,5 +167,9 @@ def get_image_name_from_url(image_url):
     return image_url[start_idx:]
 
 if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-c', '--config-file', required=True)
+    args = parser.parse_args()
+    legacy_config = get_legacy_config(args.config_file)
     config = Config.read_config(CONFIG_PATH)
-    train(config)
+    train(legacy_config, config.get("tagging_user"), config.get("url"))
```
Comment (andrebriggs) [inline train_vnext/training.py:175]: I guess I never noticed but we call the Azure Functions URL just "url"?

File: train_vnext/training.py (hunk 3 above)
Comment (samiyaakhtar) [reply train_vnext/training.py]: I'll update it to call legacy_config

File: train_vnext/training.py (hunk 5 above)
Comment (samiyaakhtar) [reply train_vnext/training.py:175]: Yeah the old config puts it into "url"

**Review by andrebriggs — APPROVED** [review]

**Pull request merged**